scripts/staging/fm_binclass.dml

```diff
@@ -0,0 +1,50 @@
+#-------------------------------------------------------------
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+# 
+#   http://www.apache.org/licenses/LICENSE-2.0
+# 
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+#-------------------------------------------------------------
+
+/*
+ * Factorization Machines for binary classification.
+ */
+
+# Imports
+source("staging/fm.dml") as fm
+source("nn/layers/sigmoid.dml") as sigmoid
+source("nn/layers/log_loss.dml") as log_loss
+
+train = function(matrix[double] X)
+    return (matrix[double] out) {
+   /*
+    * Trains a FM model.
+    */
```

Comment:
here's a basic skeleton:

* initialize the `fm` core
* choose and initialize an optimizer for `w0`, `W`, and `V`.  I would suggest [`adam`](https://github.com/apache/systemml/blob/master/scripts/nn/optim/adam.dml), calling the `init` method for each of `w0`, `W`, and `V`.
* send the inputs through `fm::forward`.
* send that result through `sigmoid::forward`
* send that through `log_loss::forward` if you want to print out the loss
* every once in a while compute loss and accuracy for the validation set and print it out
* send the result of `sigmoid::forward` and the correct labels `Y` to `log_loss::backward`
* send that through `sigmoid::backward`
* send that through `fm::backward`
* call `adam::update` for all params

use this as a reference skeleton: https://github.com/apache/systemml/blob/master/scripts/nn/examples/mnist_lenet.dml#L36